clause.go

```diff
@@ -89,30 +89,17 @@ func (c *Clause) PreAction(action Action) *Clause {
 
 // HintAction registers a HintAction (function) for the flag to provide completions
 func (c *Clause) HintAction(action HintAction) *Clause {
-	c.addHintAction(action)
+	c.addUserCompletionWords(action()...)
```

Comment:
I think this should still be a callback, not expanded at CLI configuration time.